internal/webhooks/capp_mutator_webhook.go

```diff
@@ -0,0 +1,55 @@
+package webhooks
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+
+	cappv1alpha1 "github.com/dana-team/container-app-operator/api/v1alpha1"
+
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+type CappMutator struct {
+	Decoder *admission.Decoder
+}
+
+// +kubebuilder:webhook:path=/mutate-capp,mutating=true,sideEffects=NoneOnDryRun,failurePolicy=fail,groups=rcs.dana.io,resources=capps,verbs=create;update,versions=v1alpha1,name=capp.dana.io,admissionReviewVersions=v1;v1beta1
+
+const (
+	LastUpdatedByAnnotationKey = "rcs.dana.io/last-updated-by"
+	MutatorServingPath         = "/mutate-capp"
+)
+
+// Handle implements the mutation webhook.
+func (c *CappMutator) Handle(ctx context.Context, req admission.Request) admission.Response {
+	logger := log.FromContext(ctx).WithValues("mutation webhook", "capp mutation Webhook", "Name", req.Name)
+	capp := cappv1alpha1.Capp{}
+	if err := c.Decoder.DecodeRaw(req.Object, &capp); err != nil {
+		logger.Error(err, "could not decode capp object")
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+
+	return c.handle(capp, req)
+}
+
+func (c *CappMutator) handle(capp cappv1alpha1.Capp, req admission.Request) admission.Response {
+	c.addAnnotation(&capp, req)
+
+	marshaledCapp, err := json.Marshal(capp)
```

Comment:
I think this logic can actually stay in the `Handle` function since it's mostly just boilerplate. I would have a `Handle` function with all the boilerplate stuff, and a `handleInner` function with just the annotation update logic